lib/src/main/java/growthbook/sdk/java/multiusermode/GrowthBookMultiUser.java

```diff
@@ -0,0 +1,131 @@
+package growthbook.sdk.java.multiusermode;
+
+import growthbook.sdk.java.*;
+import growthbook.sdk.java.multiusermode.configurations.EvaluationContext;
+import growthbook.sdk.java.multiusermode.configurations.GlobalContext;
+import growthbook.sdk.java.multiusermode.configurations.Options;
+import growthbook.sdk.java.multiusermode.configurations.UserContext;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+
+@Slf4j
+public class GrowthBookMultiUser {
+
+    private final Options options;
+    private final FeatureEvaluator featureEvaluator;
+    private final ExperimentEvaluator experimentEvaluatorEvaluator;
+    private static GBFeaturesRepository repository;
+    private final ArrayList<ExperimentRunCallback> callbacks;
+
+    public GrowthBookMultiUser(Options opts) {
+        this.options = opts == null ? Options.builder().build() : opts;
+
+        this.featureEvaluator = new FeatureEvaluator();
+        this.experimentEvaluatorEvaluator = new ExperimentEvaluator();
+        this.callbacks = new ArrayList<>();
+    }
+
+    public GrowthBookMultiUser() {
+        this.options = Options.builder().build();
+
+        this.featureEvaluator = new FeatureEvaluator();
+        this.experimentEvaluatorEvaluator = new ExperimentEvaluator();
+        this.callbacks = new ArrayList<>();
+    }
+
+    public void initialize() {
+        try {
+            // load features, experiments, sticky bucket thing whatever!
+            if (!this.options.getIsCacheDisabled()) {
+                // enable cache? is there anything we could do here!
+            }
+
+            if (repository == null) {
+                repository = GBFeaturesRepository.builder()
+                        .apiHost(this.options.getApiHost())
+                        .clientKey(this.options.getClientKey())
+                        .decryptionKey(this.options.getDecryptionKey())
+                        .refreshStrategy(this.options.getRefreshStrategy())
+                        .build();
+
+                // Add featureRefreshCallback
+                repository.onFeaturesRefresh(this.options.getFeatureRefreshCallback());
+
+                try {
+                    repository.initialize();
+                } catch (FeatureFetchException e) {
+                    log.error("Failed to initialize features repository", e);
+                    throw new RuntimeException(e);
+                }
+            }
+
+            // load features JSON
+            //repository.getFeaturesJson();
+
+        } catch (Exception e) {
+            log.error("Failed to initialize growthbook instance", e);
+        }
+    }
+
+    private GlobalContext getGlobalContext() {
+        GlobalContext globalContext =  GlobalContext.builder()
+            .featuresJson(repository.getFeaturesJson())
+            .build();
+
+        globalContext.loadEncryptedFeatures(this.options.getDecryptionKey());
+        return globalContext;
+    }
```

Comment:
For performance, I think we will want to only have a single `GlobalContext` object created during initialization and re-used for every evaluation.

Reply:
`GlobalContext` is not a heavy object and apart from unmarshalling the features json, there are no heavy operations. If we change the implementation to re-use the same instance over n over, we'll have to create hooks to read the latest features from the repository to override the features. And this operation would have to synchronized with any evaluations that are in-progress. Otherwise we will run into issues using this in the multi threaded context.

Reply:
The unmarshalling might be expensive for large payloads, right? 

The payload changes very infrequently (max once a minute, but usually more like hours or days), but feature evaluations might happen thousands of times per second or more.

Seems like we should only do the expensive unmarshalling part when the payload changes and not for every eval.

For background, one of the main reasons we're doing this refactor is for customers with massive request volume running into performance bottlenecks. So anything we can do to reduce work for each eval is important.

Reply:
Yeah, I found a way to optimize this in a simple way. Almost done with the changes, I'll update you!